src/SafientCore.ts

```diff
@@ -465,9 +465,15 @@ export class SafientCore {
     try {  
       const result: Safe = await getSafeData(safeId);
       // TODO: Need to move the different function
-      // if(result.decSafeKeyShards.length >=2 && result.stage !== SafeStages.CLAIMED){
-      //    result.stage = SafeStages.RECOVERED
-      // }
+      if(result.stage === SafeStages.CLAIMING){
+        const claimStatus = await this.getClaimStatus(safeId, result.claims[result.claims.length - 1].disputeId)
```

Comment:
What will this return if the `claims` is empty

Reply:
So claims won't be empty unless the safe stage is `CLAIMING`

Reply:
![Screenshot from 2022-01-17 17-19-23](https://user-images.githubusercontent.com/4023530/149764769-e56fe0ce-f5e2-4037-bacf-4caf3f0836eb.png)
Claims will be empty even in the active stage. So, this might throw an error for `claims[-1].disputeId`. Please check

Reply:
Yeah, Claims will be empty in `ACTIVE` safe stage for first time. Once the claim is created, it moves to `CLAIMING` where the claims will have claim data.